backend/src/packages/business/business.controller.ts

```diff
@@ -368,6 +376,39 @@ class BusinessController extends Controller {
           }>,
         ),
     });
+
+    this.addRoute({
+      path: BusinessApiPath.TRUCKS,
+      method: 'GET',
+      authStrategy: defaultStrategies,
+      validation: {
+        params: businessGetAllTrucksParameters,
+      },
+      handler: (options) =>
+        this.findAllTrucks(
+          options as ApiHandlerOptions<{
+            params: BusinessGetAllTrucksRequestParameters;
+            query: PaginationPayload;
+          }>,
+        ),
+    });
+
+    this.addRoute({
+      path: BusinessApiPath.TRUCKS,
+      method: 'POST',
+      authStrategy: defaultStrategies,
+      validation: {
+        body: truckCreateRequestBody,
+        params: truckCreateRequestParameters,
+      },
+      handler: (request) =>
+        this.createTruck(
+          request as ApiHandlerOptions<{
+            body: Omit<TruckEntity, 'id' | 'createdAt' | 'businessId'>;
+            params: { businessId: number };
```

Comment:
I think you don't need to pass through params business Id, as this api will be available only for authorized users, and this user should have business Id included